### PR TITLE
feature: add verbose logging for clone/fetch command

### PR DESCRIPTION
**Pull request by rslabbert**
Builds on #771

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -58,6 +58,8 @@ slab = "0.4.7"
 tempfile = "3.3.0"
 textwrap = "0.16.0"
 thiserror = "1.0.37"
+tracing = "0.1.37"
+tracing-subscriber = "0.3.16"
```
Comment (Ralith) [inline Cargo.toml:62]: tracing-subscriber's default-features brings in a lot, as indicated by the 107-line Cargo.lock diff. Let's `default-features = false` this and enable only the minimally needed stuff.

**Review by Ralith — DISMISSED** [review]

File: Cargo.toml (hunk 1 above)
Comment (rslabbert) [reply Cargo.toml]: Updated with std, fmt, and ansi

**Comment by martinvonz** [comment]
This is meant to have `rslabbert:push-01c93fd5ebc946a5b71ff2acbb1a50e9` as parent, right? We have #485 for setting that up automatically, but for now you have to do it manually

**Review by martinvonz — APPROVED** [review]
Thanks!

File: lib/src/git.rs [hunk 2]
```diff
@@ -553,38 +560,67 @@ impl<'a> RemoteCallbacks<'a> {
...
+                                },
+                            );
                         }
                     }
                 }
                 if allowed_types.contains(git2::CredentialType::USER_PASS_PLAINTEXT) {
                     if let Some(ref mut cb) = self.get_password {
                         if let Some(pw) = cb(url, username) {
-                            return git2::Cred::userpass_plaintext(username, &pw);
+                            tracing::debug!(username, "using userpass_plaintext");
```
Comment (Ralith) [inline lib/src/git.rs:604]: Could note here that we're getting the username from the URL.

File: src/commands.rs [hunk 3]
```diff
@@ -3968,6 +3968,7 @@ fn cmd_git_remote_list(
     Ok(())
 }
 
+#[tracing::instrument(skip(ui, command))]
```
Comment (Ralith) [inline src/commands.rs:3971]: Is it useful to log args here? The user presumably knows/can see what they passed.

Comment (rslabbert) [reply src/commands.rs:3971]: I was going with the approach of logging anything that had a debug impl. Having the args running through into verbose logs might also help if a user is sharing only their logs for output and not their command.

File: lib/src/git.rs (hunk 2 above)
Comment (rslabbert) [reply lib/src/git.rs]: Added!

**Pull request merged**